File: Two_Pointers/sol.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:

        left = 0
        right = len(nums) - 1
        result = []
        while left <= right:
            if nums[left]**2 > nums[right]**2:
                result.insert(0, nums[left]**2)
                left +=1
            else:
                result.insert(0, nums[right]**2)
                right -= 1
        return result

File: SlidingWindow/1493. Longest Subarray of 1's After Deleting One Element/sol.ts
function longestSubarray(nums: number[]): number {
  let left = 0;
  let right = 0;

  let best = 0;
  let deletedZero = false;

  while (right < nums.length) {
    if (nums[right] === 0) {
      if (deletedZero) {
        while (deletedZero) {
          if (nums[left] === 0) {
            deletedZero = false;
          }
          left++;
        }
      }
      deletedZero = true;
    }
    best = Math.max(right - left, best);
    right++;
  }
  return best;
}
